airflow/plugins/operators/scrape_state_geoportal.py

```diff
@@ -0,0 +1,206 @@
+    root_url: str
+    service: str
+    product: str
+    where: str
+    outFields: str
+    f: str
+    resultRecordCount: int
+
+    @property
+    def table(self) -> str:
+        return self.product
+
+    @property
+    def filename(self) -> str:
+        return self.table
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    def fetch_from_state_geoportal(self):
+        """ """
+
+        logging.info(f"Downloading state geoportal data for {self.product}.")
+
+        try:
+            # Set up the parameters for the request
+            url = self.root_url + self.service
+            validated_url = parse_obj_as(HttpUrl, url)
+
+            params = {
+                "where": self.where,  # You can change this to filter data
+                "outFields": self.outFields,  # Specify the fields to return
+                "f": self.f,  # Format of the response
+                "resultRecordCount": self.resultRecordCount,  # Maximum number of rows per request
+            }
+
+            all_features = []  # To store all retrieved rows
+            offset = 0
+
+            while True:
+                # Update the resultOffset for each request
+                params["resultOffset"] = offset
+
+                # Make the request
+                response = requests.get(validated_url, params=params)
+                # response = requests.get(url, params=params)
+                data = response.json()
+
+                # Break the loop if there are no more features
+                if "features" not in data or not data["features"]:
+                    break
+
+                # Append the retrieved features
+                all_features.extend(data["features"])
+
+                # Increment the offset
+                offset += params["resultRecordCount"]
+
+            if all_features is None or len(all_features) == 0:
+                logging.info(
+                    f"There is no data to download for {self.product}. Ending pipeline."
+                )
+
+                pass
+            else:
+                logging.info(
+                    f"Downloaded {self.product} data with {len(all_features)} rows!"
+                )
+
+                return all_features
+
+        except requests.exceptions.RequestException as e:
+            logging.info(f"An error occurred: {e}")
+
+            raise
+
+
+# # Function to convert coordinates to WKT format
+def to_wkt(geometry_type, coordinates):
+    if geometry_type == "LineString":
+        # Format as a LineString
+        coords_str = ", ".join([f"{lng} {lat}" for lng, lat in coordinates])
+        return f"LINESTRING({coords_str})"
+    elif geometry_type == "MultiLineString":
+        # Format as a MultiLineString
+        multiline_coords_str = ", ".join(
+            f"({', '.join([f'{lng} {lat}' for lng, lat in line])})"
+            for line in coordinates
+        )
+        return f"MULTILINESTRING({multiline_coords_str})"
+    else:
+        return None
```

Comment:
**question:** Should work. Is this from somewhere online or original?

Reply:
mostly found it online and added some tweaks